### DISCUSSION TITLE
采样组件运行失败

### DISCUSSION THREAD

**Discussion by 1998zxn** [secretflow/secretflow#1524] [Q&A] [answered]
任务流如下：
![image](https://github.com/user-attachments/assets/5b1fa86e-2df2-47a3-8d45-4494aaa20dd4)
样本表输入为：
[泰坦尼克号遇难名单](https://www.kaggle.com/competitions/titanic/data)的训练集。
报错提示：
`2024-10-09 09:17:38,135|boole-hpc-01|ERROR|secretflow|entry.py:<module>:601| unexpected exception
Traceback (most recent call last):
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/entry.py\", line 599, in <module>
    main()
  File \"/usr/local/lib/python3.10/site-packages/click/core.py\", line 1157, in __call__
    return self.main(*args, **kwargs)
  File \"/usr/local/lib/python3.10/site-packages/click/core.py\", line 1078, in main
    rv = self.invoke(ctx)
  File \"/usr/local/lib/python3.10/site-packages/click/core.py\", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File \"/usr/local/lib/python3.10/site-packages/click/core.py\", line 783, in invoke
    return __callback(*args, **kwargs)
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/entry.py\", line 561, in main
    sf_node_eval_param = preprocess_sf_node_eval_param(
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/entry.py\", line 328, in preprocess_sf_node_eval_param
    domaindata_id_to_dist_data(
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/entry.py\", line 191, in domaindata_id_to_dist_data
    download_dist_data_from_dp(
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/entry.py\", line 142, in download_dist_data_from_dp
    get_file_from_dp(
  File \"/usr/local/lib/python3.10/site-packages/secretflow/kuscia/datamesh.py\", line 192, in get_file_from_dp
    for batch in flight_reader:
  File \"pyarrow/ipc.pxi\", line 666, in pyarrow.lib.RecordBatchReader.__next__
  File \"pyarrow/ipc.pxi\", line 700, in pyarrow.lib.RecordBatchReader.read_next_batch
  File \"pyarrow/error.pxi\", line 91, in pyarrow.lib.check_status
pyarrow.lib.ArrowException: Unknown error: Flight RPC failed with message: strconv.ParseInt: parsing \"\": invalid syntax. gRPC client debug context: UNKNOWN:Error received from peer ipv4:172.18.0.3:8071 {created_time:\"2024-10-09T09:17:38.134772114+00:00\", grpc_status:2, grpc_message:\"strconv.ParseInt: parsing \\\"\\\": invalid syntax\"}. Client context: OK. Detail: Failed
"`
本地随意构造的数据集作为样本表的输入，反而能够正确执行。
![image](https://github.com/user-attachments/assets/68fba44a-fd1b-4779-8648-86fa91ba3d7e)
看报错并不是由于数据规范导致失败，采样组件是否对数据有一定的要求？

**Comment by 1998zxn** [comment, marked as answer]
数据集中存在空值无法处理。后续还有一个错误是自动设别字段时识别错误，float字段类型设置为int导致失败。

**Reply by zimu-yuxi** [reply]
使用的是哪个版本呢？

**Reply by 1998zxn** [reply]
![image](https://github.com/user-attachments/assets/de58f33c-0191-4c81-ab2a-856b26a91eb7)

**Reply by zimu-yuxi** [reply]
嗯～～为什么不在数据对齐后再做采样呢。是否考虑过随机采样后求交结果可能为空呢

**Reply by 1998zxn** [reply]
为什么不在数据对齐后再做采样呢==》从我的角度讲，我之前并没有很多相关经验，同时没有看到组件对输入有做相关要求（例如规范化、数据对齐等），所以直接将原始数据放了进去。
是否考虑过随机采样后求交结果可能为空呢==》对于随机采样我也没有做过了解，求交是随机采样后必须的步骤吗？当然我也在后面设置了隐私求交来获取垂直表，不过提示assert device.world_size == 2, 'only 2pc is allowed.'断言失败，还没有去调试。
